tests/test_test.py

```diff
@@ -0,0 +1,78 @@
+from pathlib import Path
+from _pytest.capture import CaptureFixture
+from _pytest.monkeypatch import MonkeyPatch
+
+import rasa.model
+import rasa.cli.utils
+
+
+def monkeypatch_get_latest_model(tmp_path: Path, monkeypatch: MonkeyPatch):
+
+    latest_model = tmp_path / "my_test_model.tar.gz"
+
+    def mock_get_latest_model():
+        return str(latest_model)
+
+    monkeypatch.setattr(rasa.model, "get_latest_model", mock_get_latest_model)
+
+
+def test_test_core_models_in_directory_input_default(
+    capsys: CaptureFixture, tmp_path: Path, monkeypatch: MonkeyPatch
+):
+    from rasa.test import _test_core_models_in_directory_input
+
+    monkeypatch_get_latest_model(tmp_path, monkeypatch)
+
+    latest_model = Path(rasa.model.get_latest_model())
+    latest_model.touch()
+
+    # Input: default model file
+    # => Should return containing directory
+    modeldir = rasa.model.get_latest_model()
+    p = Path(modeldir)
+    new_modeldir = _test_core_models_in_directory_input(modeldir)
+    captured = capsys.readouterr()
+    assert captured.out == ""
+    assert new_modeldir == str(p.parent)
+
+
+def test_test_core_models_in_directory_input_file(
+    capsys: CaptureFixture, tmp_path: Path, monkeypatch: MonkeyPatch
+):
+    from rasa.test import _test_core_models_in_directory_input
+
+    monkeypatch_get_latest_model(tmp_path, monkeypatch)
```

Comment:
I think it would be better to path the model name in `monkeypatch_get_latest_model` instead of `tmp_path` because each test case then decide on the model name on it's own and you avoid using hardcoded strings like `my_test_model1.tar.gz` in both functions

Reply:
At the moment, I'm only hardcoding the string in one place: here in `monkeypatch_get_latest_model`. All the test cases then just call rasa.model.get_latest_model() to get the same string.

In my opinion, all test cases using the same model name is a good thing. If we have to pass the model name instead and each test case could use a different one, what would be the advantage? I'd say it only makes the code harder to read bc. model names are defined independently (hardcoded) in several places. What do you think?